### PR TITLE
[Service Bus] Clarifiy ScheduledEnqueueTime

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to clarify how scheduled messages work.

# References and Resources

- [Clarify ScheduledEnqueueTime (#18588)](https://github.com/Azure/azure-sdk-for-net/issues/18588)

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs [hunk 1]
```diff
@@ -380,17 +380,13 @@ public string ReplyTo
         }
 
         /// <summary>
-        /// Gets or sets the date and time in UTC at which the message will be enqueued. This
-        /// property returns the time in UTC; when setting the property, the supplied DateTime value must also be in UTC.
+        /// Gets or sets the date and time, in UTC, at which the message should be be made availible to receivers. This property does not control when a message is sent by the 
+        /// client.  Sending happens immediately when `SendAsync` is called.  Service Bus will hide the message from receivers until the the requested time.
```
Comment (christothes) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs:384]:
```suggestion
        /// Gets or sets the date and time, in UTC, at which the message should be made available to receivers. This property does not control when a message is sent by the
        /// client.  Sending happens immediately when `SendAsync` is called.  Service Bus will hide the message from receivers until the the requested time.
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs [hunk 2]
```diff
@@ -380,17 +380,13 @@ public string ReplyTo
...
         /// </summary>
         /// <value>
-        /// The scheduled enqueue time in UTC. This value is for delayed message sending.
-        /// It is utilized to delay messages sending to a specific time in the future.
+        /// The date and time, in UTC, that the message should be available to receivers.  This time may not be exact, the actual time depends on the entity's workload and state.
```
Comment (christothes) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs:387]:
```suggestion
        /// The date and time, in UTC, at which the message should be available to receivers.  This time may not be exact, the actual time depends on the entity's workload and state.
```

**Review by christothes — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs [hunk 3]
```diff
@@ -380,17 +380,13 @@ public string ReplyTo
         }
 
         /// <summary>
-        /// Gets or sets the date and time in UTC at which the message will be enqueued. This
-        /// property returns the time in UTC; when setting the property, the supplied DateTime value must also be in UTC.
+        /// Gets or sets the date and time, in UTC, at which the message should be be made availible to receivers. This property does not control when a message is sent by the
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs:383]:
```suggestion
        /// Gets or sets the date and time, in UTC, at which the message should be be made available to receivers. This property does not control when a message is sent by the 
```

**Comment by JoshLove-msft** [comment]
Should we also update https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebusreceivedmessage.scheduledenqueuetime?view=azure-dotnet?

**Comment by jsquire** [comment]
> Should we also update https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebusreceivedmessage.scheduledenqueuetime?view=azure-dotnet?

Good call.  I forgot that there was no inheritance relationship.   Received message has been updated.

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs [hunk 4]
```diff
@@ -380,17 +380,13 @@ public string ReplyTo
...
         /// <summary>
-        /// Gets or sets the date and time in UTC at which the message will be enqueued. This
-        /// property returns the time in UTC; when setting the property, the supplied DateTime value must also be in UTC.
+        /// Gets or sets the date and time, in UTC, at which the message should be be made available to receivers. This property does not control when a message is sent by the 
+        /// client.  Sending happens immediately when `SendAsync` is called.  Service Bus will hide the message from receivers until the the requested time.
         /// </summary>
         /// <value>
-        /// The scheduled enqueue time in UTC. This value is for delayed message sending.
-        /// It is utilized to delay messages sending to a specific time in the future.
+        /// The date and time, in UTC, at which the message should be available to receivers.  This time may not be exact, the actual time depends on the entity's workload and state.
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs:387]:
```suggestion
        /// The date and time, in UTC, at which the message should be available to receivers.  This time may not be exact; the actual time depends on the entity's workload and state.
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs [hunk 5]
```diff
@@ -380,17 +380,13 @@ public string ReplyTo
         }
 
         /// <summary>
-        /// Gets or sets the date and time in UTC at which the message will be enqueued. This
-        /// property returns the time in UTC; when setting the property, the supplied DateTime value must also be in UTC.
+        /// Gets or sets the date and time, in UTC, at which the message should be be made available to receivers. This property does not control when a message is sent by the
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs:383]:
```suggestion
        /// Gets or sets the date and time, in UTC, at which the message should be made available to receivers. This property does not control when a message is sent by the
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusReceivedMessage.cs [hunk 6]
```diff
@@ -196,12 +196,14 @@ public static ServiceBusReceivedMessage FromAmqpMessage(AmqpAnnotatedMessage mes
...
         public string ReplyTo => AmqpMessage.Properties.ReplyTo?.ToString();
 
-        /// <summary>Gets the date and time in UTC at which the message will be enqueued. This
-        /// property returns the time in UTC; when setting the property, the supplied DateTime value must also be in UTC.</summary>
-        /// <value>The scheduled enqueue time in UTC. This value is for delayed message sending.
-        /// It is utilized to delay messages sending to a specific time in the future.</value>
-        /// <remarks> Message enqueuing time does not mean that the message will be sent at the same time. It will get enqueued, but the actual sending time
-        /// depends on the queue's workload and its state.</remarks>
+        /// <summary>
+        /// Gets or sets the date and time, in UTC, at which the message should be be made available to receivers. This property does not control when a message is sent by the
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusReceivedMessage.cs:200]:
```suggestion
        /// Gets or sets the date and time, in UTC, at which the message should be made available to receivers. This property does not control when a message is sent by the
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs [hunk 7]
```diff
@@ -380,17 +380,13 @@ public string ReplyTo
         }
 
         /// <summary>
-        /// Gets or sets the date and time in UTC at which the message will be enqueued. This
-        /// property returns the time in UTC; when setting the property, the supplied DateTime value must also be in UTC.
+        /// Gets or sets the date and time, in UTC, at which the message should be made available to receivers. This property does not control when a message is sent by the
+        /// client.  Sending happens immediately when `SendAsync` is called.  Service Bus will hide the message from receivers until the the requested time.
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs:384]:
```suggestion
        /// client. Sending happens immediately when `SendAsync` is called.  Service Bus will hide the message from receivers until the the requested time.
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs [hunk 8]
```diff
@@ -380,17 +380,13 @@ public string ReplyTo
...
         /// </summary>
         /// <value>
-        /// The scheduled enqueue time in UTC. This value is for delayed message sending.
-        /// It is utilized to delay messages sending to a specific time in the future.
+        /// The date and time, in UTC, at which the message should be available to receivers.  This time may not be exact; the actual time depends on the entity's workload and state.
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs:387]:
```suggestion
        /// The date and time, in UTC, at which the message should be available to receivers. This time may not be exact; the actual time depends on the entity's workload and state.
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusReceivedMessage.cs [hunk 9]
```diff
@@ -196,12 +196,14 @@ public static ServiceBusReceivedMessage FromAmqpMessage(AmqpAnnotatedMessage mes
...
 
-        /// <summary>Gets the date and time in UTC at which the message will be enqueued. This
-        /// property returns the time in UTC; when setting the property, the supplied DateTime value must also be in UTC.</summary>
-        /// <value>The scheduled enqueue time in UTC. This value is for delayed message sending.
-        /// It is utilized to delay messages sending to a specific time in the future.</value>
-        /// <remarks> Message enqueuing time does not mean that the message will be sent at the same time. It will get enqueued, but the actual sending time
-        /// depends on the queue's workload and its state.</remarks>
+        /// <summary>
+        /// Gets or sets the date and time, in UTC, at which the message should be made available to receivers. This property does not control when a message is sent by the
+        /// client.  Sending happens immediately when `SendAsync` is called.  Service Bus will hide the message from receivers until the the requested time.
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusReceivedMessage.cs:201]:
```suggestion
        /// client. Sending happens immediately when `SendAsync` is called.  Service Bus will hide the message from receivers until the the requested time.
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusReceivedMessage.cs [hunk 10]
```diff
@@ -196,12 +196,14 @@ public static ServiceBusReceivedMessage FromAmqpMessage(AmqpAnnotatedMessage mes
...
+        /// </summary>
+        /// <value>
+        /// The date and time, in UTC, at which the message should be available to receivers.  This time may not be exact; the actual time depends on the entity's workload and state.
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusReceivedMessage.cs:204]:
```suggestion
        /// The date and time, in UTC, at which the message should be available to receivers. This time may not be exact; the actual time depends on the entity's workload and state.
```

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**